core/src/components/input/input.tsx

```diff
@@ -260,12 +261,25 @@ export class Input implements ComponentInterface {
     }
   }
 
+  componentDidLoad() {
+    if (this.nativeInput) {
+      // TODO: Update to JSX bindings when Stencil resolves bug with:
+      // https://github.com/ionic-team/stencil/issues/3235
+      this.nativeInput.addEventListener('compositionstart', this.onCompositionStart);
+      this.nativeInput.addEventListener('compositionend', this.onCompositionEnd);
+    }
+  }
+
   disconnectedCallback() {
     if (Build.isBrowser) {
       document.dispatchEvent(new CustomEvent('ionInputDidUnload', {
         detail: this.el
       }));
     }
+    if (this.nativeInput) {
+      this.nativeInput.removeEventListener('compositionstart', this.onCompositionStart);
+      this.nativeInput.removeEventListener('compositionEnd', this.onCompositionEnd);
+    }
```

Comment:
Minor thing, but requesting `this.nativeInput` causes some extra code to be run in Stencil to get the value, so caching the value allows us to skip that code from running additional times.

```suggestion
    const nativeInput = this.nativeInput;
    if (nativeInput) {
      nativeInput.removeEventListener('compositionstart', this.onCompositionStart);
      nativeInput.removeEventListener('compositionEnd', this.onCompositionEnd);
    }
```

Reply:
Also if we can do the same thing for the addEventListener above that'd be great

Reply:
Can do 👍  I wasn't aware that accessing private members in Stencil had an overhead. Is that because it's bound to a `ref` or just in general?

Reply:
I believe anything that is declared as a variable on the component instance has this slight overhead.

So basically anything like this https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/input/input.tsx#L21-L25